test/blackbox-tests/test-cases/pkg/opam-var/opam-var-pkg.t

```diff
@@ -0,0 +1,232 @@
+  [1]
+_:depends
+  $ test _:depends
+  (build
+   (run echo %{pkg-self:depends}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-30:
+  4 |  (run echo %{pkg-self:depends}))
+                 ^^^^^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:installed
+  $ test _:installed
+  (build
+   (run echo %{pkg-self:installed}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-32:
+  4 |  (run echo %{pkg-self:installed}))
+                 ^^^^^^^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:enable
+  $ test _:enable
+  (build
+   (run echo %{pkg-self:enable}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-29:
+  4 |  (run echo %{pkg-self:enable}))
+                 ^^^^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:pinned
+  $ test _:pinned
+  (build
+   (run echo %{pkg-self:pinned}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-29:
+  4 |  (run echo %{pkg-self:pinned}))
+                 ^^^^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:bin
+  $ test _:bin
+  (build
+   (run echo %{pkg-self:bin}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-26:
+  4 |  (run echo %{pkg-self:bin}))
+                 ^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:sbin
+  $ test _:sbin
+  (build
+   (run echo %{pkg-self:sbin}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-27:
+  4 |  (run echo %{pkg-self:sbin}))
+                 ^^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:lib
+  $ test _:lib
+  (build
+   (run echo %{pkg-self:lib}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-26:
+  4 |  (run echo %{pkg-self:lib}))
+                 ^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:man
+  $ test _:man
+  (build
+   (run echo %{pkg-self:man}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-26:
+  4 |  (run echo %{pkg-self:man}))
+                 ^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:doc
+  $ test _:doc
+  (build
+   (run echo %{pkg-self:doc}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-26:
+  4 |  (run echo %{pkg-self:doc}))
+                 ^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:share
+  $ test _:share
+  (build
+   (run echo %{pkg-self:share}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-28:
+  4 |  (run echo %{pkg-self:share}))
+                 ^^^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:etc
+  $ test _:etc
+  (build
+   (run echo %{pkg-self:etc}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-26:
+  4 |  (run echo %{pkg-self:etc}))
+                 ^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:build
+  $ test _:build
+  (build
+   (run echo %{pkg-self:build}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-28:
+  4 |  (run echo %{pkg-self:build}))
+                 ^^^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+_:hash
```

Comment:
All explicitly unsupported variables like this one can be moved to their own test file

Reply:
The `pkg-self` macro not working doesn't tell me if we are not intending to support it. So I'm not sure exactly which ones we are intending to support. Is there a list somewhere?

Reply:
I've moved `root` and `_:hash` to an unsupported variables file anyway.